inout.go

```diff
@@ -163,6 +163,26 @@ import "go.uber.org/dig"
 // Note that values in a value group are unordered. Fx makes no guarantees
 // about the order in which these values will be produced.
 //
+// To declare a soft relationship between a group an its constructors, use the
+// `soft` option on the group tag (`group:"[groupname],soft"`). A soft group
+// will be populated only with values from already-executed constructors, this
+// means constructors of soft value groups will be called only if they provide
+// another requested value.
+//
+//   type Params struct {
+//	  fx.In
+//
+//    Handlers []Handler `group:"server"`
+//    Logger   *zap.Logger
+//   }
+//   func NewHandlerAndLogger() (Handler, *zap.Logger) { ... }
+//   func NewHandler() Handler { ... }
+//   func Foo(Params) { ... }
+//
+// When `NewHandlerAndLogger` and `NewHandler`are provided, and `Foo` invoked,
+// the only constructor called is `NewHandler`, because this also provides
+// `*zap.Logger` needed in the `Params` struct.
+//
```

Comment:
Apart from this example, can you also provide an example where:

1. If you have a constructor for a type, and the only consumer is the soft value group, it is not populated. 
2. In the exact same case above, add a consumer that also consumes that type. Then show that it is populated.